### ISSUE TITLE
Distribute strategy(scaling loss per_replica)?

### ISSUE THREAD

**Issue by capilano** [captain-pool/GSOC#57]
Are you scaling the loss when using distribute strategy, or are you using the same setup as training using estimators?

**Comment by captain-pool** [comment]
> Are you scaling the loss when using distribute strategy, or are you using the same setup as training using estimators?

Hey @capilano, yes, I'm scaling the loss per replica. https://github.com/captain-pool/GSOC/blob/37f442fe3254218cd99755a2c77bf7ba862e0a6c/E2_ESRGAN/lib/train.py#L113

**Comment by capilano** [comment]
I see, Thanks. So you are dividing by batch size  and not by the no of replicas?

**Comment by captain-pool** [comment]
> I see, Thanks. So you are dividing by batch size and not by the no of replicas?
> Edit: So I guess you do tf.reduce_sum instead of tf.reduce_mean and then divide by the global batch size?

Yup! Here, check this out for further reference. https://www.tensorflow.org/tutorials/distribute/training_loops#define_the_loss_function

**Comment by capilano** [comment]
Yeah,I have looked at that. Its the same as doing tf.reduce_mean and then dividing by the number of synced replicas in the TPU case,atleast
